compiler/tests-common/tests/org/jetbrains/kotlin/codegen/ir/AbstractIrLineNumberTest.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.codegen.ir
+
+import org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment
+import org.jetbrains.kotlin.codegen.AbstractLineNumberTest
+import org.jetbrains.kotlin.codegen.ClassFileFactory
+import org.jetbrains.kotlin.config.JVMConfigurationKeys
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.test.testFramework.KtUsefulTestCase
+import org.jetbrains.org.objectweb.asm.ClassReader
+
+abstract class AbstractIrLineNumberTest : AbstractLineNumberTest() {
+    override fun createEnvironment(setup: Boolean): KotlinCoreEnvironment {
+        val environment = super.createEnvironment(setup)
+        if (!setup) {
+            // Compile setup class always with original non-IR backend
+            environment.configuration.put(JVMConfigurationKeys.IR, true)
```

Comment:
Use updateConfiguration/extractConfigurationKind to configure ir stuff (rebase is required)